### PR TITLE
fix(forms): validate required bridge message fields (MAGE-484)

### PR DESCRIPTION
**Pull request by evan-masseau**
# Description

Adds validation to form lifecycle event parsing at the native-to-JS bridge boundary. When mapping raw native event payloads to the TypeScript `FormLifecycleEvent` type, required fields (`type`, `formId`, `formName`, and `buttonLabel` for CTA events) are now validated. Malformed events are logged and dropped rather than forwarded to the host app's callback.

Part of MAGE-484

## Due Diligence

- [ ] I have tested this on a simulator/emulator or a physical device, on iOS and Android (if applicable).
- [x] I have added sufficient unit/integration tests of my changes.
- [ ] I have adjusted or added new test cases to team test docs, if applicable.
- [x] I am confident these changes are implemented with feature parity across iOS and Android (if applicable).

## Release/Versioning Considerations

- [x] `Patch` Contains internal changes or backwards-compatible bug fixes.
- [ ] `Minor` Contains changes to the public API.
- [ ] `Major` Contains **breaking** changes.
- [ ] Contains readme or migration guide changes.
- [x] This is planned work for an upcoming release.

## Changelog / Code Overview

- **`src/Forms.ts`**: Added `parseFormLifecycleEvent()` — a parse function that validates required fields and returns `null` for invalid payloads, logging an error with the specific missing fields.
- **`src/index.tsx`**: Updated the `NativeEventEmitter` listener in `registerFormLifecycleHandler` to use the new parser instead of an unsafe `as` cast. Invalid events are silently dropped (after logging).
- **`src/__tests__/Forms.test.ts`**: New dedicated test file for the parser with 18 test cases covering valid events, invalid types, missing/empty required fields, and edge cases.
- **`src/__tests__/index.test.tsx`**: Added 8 integration tests verifying that invalid payloads don't reach the host app handler.

## Test Plan

- `yarn test` — 54 tests pass (26 new)
- `yarn lint` — clean (pre-existing Swift warnings only)
- `yarn typecheck` — clean

## Related Issues/Tickets

Part of MAGE-484

### CODE REVIEW

File: src/Forms.ts [hunk 1]
```diff
@@ -72,3 +72,91 @@ export type FormLifecycleEvent =
...
+    );
+    return null;
+  }
+
+  const missingFields: string[] = [];
+  if (!isNonEmptyString(formId)) missingFields.push('formId');
+  if (!isNonEmptyString(formName)) missingFields.push('formName');
+
+  if (type === 'formCtaClicked') {
+    if (typeof data.buttonLabel !== 'string') missingFields.push('buttonLabel');
```
[the remark this answers is not in the document]
Comment (evan-masseau) [reply src/Forms.ts]: yep, decided to allow those.

File: src/Forms.ts [hunk 2]
```diff
@@ -72,3 +83,91 @@ export type FormLifecycleEvent =
...
+  const { type, formId, formName } = data;
+
+  if (
+    !isNonEmptyString(type) ||
+    !FORM_LIFECYCLE_EVENT_TYPES.includes(
+      type as (typeof FORM_LIFECYCLE_EVENT_TYPES)[number]
+    )
+  ) {
+    console.warn(
+      `[Klaviyo] Ignoring form lifecycle event with invalid type: ${JSON.stringify(type)}`
```
Comment (evan-masseau) [inline src/Forms.ts:126]: realistically, the native SDKs catch these. but this adds safety to _this_ SDK for proper defensiveness

**Review by josephzhou-klaviyo — APPROVED** [review]
lgtm

**Review by ajaysubra — APPROVED** [review]
LGTM
